### ISSUE TITLE
generate_svn_version.sh fails when building 7.0.65 (and earlier)

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#1225]
**Reported by Bluefin Tuna on 6 Aug 43381293 06:12 UTC**
In the "generate_svn_version.sh" there are these instructions:

```
BRANCH=`git branch | sed -n 's/^\* *//p'`
remote=`git config --get branch.$BRANCH.remote`
```

Execution of the script fails because quoting is missing. Debugging shows:

```
BRANCH='(no branch)'
git config --get 'branch.(no' 'branch).remote'
```

The above fails with 

```
error: invalid key: branch.(no
```

In this case, 'git branch' shows:

$ git branch
- (no branch)
  master

[It is unclear why these instructions are being run. The output for "git branch" should show the local branch, which should always be the "master" branch unless the user has decided to create a local branch, but that branch would not have any relationship to a branch on the git repository](Remark:)

Solution: Change script to read

```
BRANCH=`git branch | sed -n 's/^\* *//p'`
remote=`git config --get 'branch.$BRANCH.remote'`
```

One can assume that $BRANCH does not itself contain single quotes...

Migrated-From: http://boinc.berkeley.edu/trac/ticket/1253

**Comment by romw** [comment]
**Commented by Bluefin Tuna on 19 Sep 43440552 00:59 UTC**
Still present in 7.1.11

**Comment by romw** [comment]
**Commented by Bluefin Tuna on 29 Jun 44108801 02:16 UTC**
Still present in 7.2.9

**Comment by romw** [comment]
**Commented by Bluefin Tuna on 22 Dec 45086372 02:17 UTC**
In 7.4.41 the shell code has changed but the problem is still present:

```
   GIT_LOG=`git log -n1 --pretty="format:%H"`
   HOST=`hostname`
   BRANCH=`git branch | sed -n 's/^\* *//p'`
   remote=`git config --get branch.$BRANCH.remote` <------- PROBLEM WAITING TO HAPPEN
   URL=`git config --get remote.$remote.url`
   DATE=`git log -n1 --pretty="format:%ct"`
```

Tracing this yields:

```
+ BRANCH='(detached from client_release/7.4/7.4.41)'
++ git config --get 'branch.(detached' from 'client_release/7.4/7.4.41).remote'
```

So make fails:

```
error: wrong number of arguments
usage: git config [the call to the git command by adding double quotes:

```
if [ -d .git/svn ](options]
```

*FIX:*

Change); then
    CMD="git svn info"
elif [ -d .git ]; then
    GIT_LOG=`git log -n1 --pretty="format:%H"`
    HOST=`hostname`
    BRANCH=`git branch | sed -n 's/^\* *//p'`
    remote=`git config --get "branch.$BRANCH.remote"`
    URL=`git config --get "remote.$remote.url"`
    DATE=`git log -n1 --pretty="format:%ct"`
elif [ -d .svn ]; then
    CMD="svn info"
else
    CMD=""
fi
```

Setting priority to "major" because:

1) Builds fail consistently since like forever
2) And this is so easy to fix.... come ON, guys!

**Comment by ChristianBeer** [comment]
Fixed in 59faa92e1e27418ba194c70384fa7e5e20a2f116 but not backported to release branches so I'm keeping this open. we can't fix old releases but should fix new releases.

**Comment by AenBleidd** [comment]
This fix is already present in last release (and some previous) and will be present in all future releases due to our current release policy.
@ChristianBeer, can this issue be closed now?
